deps/GraphBLAS/Source/GxB_Matrix_tuple_iter.c

```diff
@@ -183,10 +377,14 @@ GrB_Info GxB_MatrixTupleIter_reset
 ) {
 	GB_WHERE1("GxB_MatrixTupleIter_reset (iter)") ;
 	GB_RETURN_IF_NULL(iter) ;
-	iter->nnz_idx  =  0              ;
-	iter->row_idx  =  0              ;
-	iter->p        =  iter->A->p[0]  ;
-	return (GrB_SUCCESS) ;
+	if(iter->iterators_size == 0) return GrB_INVALID_VALUE ;
+	iter->iterators_idx = 0;
+
+	GrB_Info rv = GrB_SUCCESS ;
+	for(size_t i = 0; i < iter->iterators_size; i++)
+		rv |= GxB_MatrixTupleIter_init(&iter->iterators[i], iter->iterators[i].A) ;
```

Comment:
I think we should break and return failure if `GxB_MatrixTupleIter_init` doesn't return `GrB_SUCCESS`